nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -191,7 +193,9 @@ public DocumentModel createComment(DocumentModel documentModel, DocumentModel co
 
             commentModelToCreate = session.createDocument(commentModelToCreate);
             commentModelToCreate.detach(true);
-            notifyEvent(session, CommentEvents.COMMENT_ADDED, documentModel, commentModelToCreate);
+            DocumentModel topLevelDocument = session.getDocument(
+                    getTopLevelCommentAncestor(session, commentModelToCreate.getRef()));
+            notifyEvent(session, CommentEvents.COMMENT_ADDED, topLevelDocument, commentModelToCreate);
             return commentModelToCreate;
         });
```

Comment:
You cannot use `getTopLevelCommentAncestor` for each comment / annotation creation, this method will returns the first commented document (File in most cases). let assume we have the comment tree as below 
`File1` <- `Comment1` <- `Reply1` <- `Reply2`...
when you add `Comment1` the commented document is `File1`
but in the case of `Reply2` in the email notification it should be `Reply1` but using `getTopLevelCommentAncestor` will give `File1` and it will breaking the existing behaviour (Jira: _Surprisingly, it works well for second level replies_).

you can recheck with the PO of the needed behaviour, but i think it will more reusable code if we introduce a new method like `getCommentedDocument` which will be implemented by `TreeCommentManager` and `PropertyCommentManager` and this method will be called we you call the notification, the `notify` method works well, i don't think we need to change his behaviour but the parameter of commentedDoc that the callers give is the wrong one.

Reply:
I checked it with @aescaffre, he wants the top level comment always. We don't want to link a comment as the source of the Notification as it does'nt have a dedicated view.

Reply:
ah ok i see, perhaps we should update the jira ticket to have this needed behaviour. then i was thinking about refactoring this part of the code : 

 ```
DocumentModel topLevelDocument = session.getDocument(
                    getTopLevelCommentAncestor(session, commentDocModel.getRef()));
            notifyEvent(session, CommentEvents.COMMENT_ADDED, topLevelDocument, commentDocModel);
```

we did the same many times in the code perhaps we can extract a new method `notifyEvent(CoreSession session, DocumentModel commentDocModel, String eventType` and this method will call the exiting one after computing the top level comment wdyt, the idea is to have a small method on comment ...

Reply:
rebased on 28253

Reply:
It has been decided to keep the 4 args method when creating comments